### PR TITLE
GH-809 Add integration tests to PR/release actions

### PR DESCRIPTION
**Pull request by jack-r-warren**
### Purpose

- https://broadinstitute.atlassian.net/browse/GH-809

### Changes

- Uses Tom's code from [his branch](https://github.com/broadinstitute/push-to-cloud-service/tree/tbl/GH-828-jms) to fix the authentication to Google Cloud (I branched off of his to try to avoid merge issues)
- Adds code to both the release and pr-test actions to set up Vault and Google Cloud
  - Lifted from https://github.com/broadinstitute/gotc-deploy/blob/master/.github/workflows/TerraformDeployTest-gotc-sa-non-prod.yml
  - Both jobs now expect four GitHub secrets:
    - ROLE_ID
    - SECRET_ID
    - VAULT_ADDR
    - ~~SA_KEY_PATH (actual path to the service account key .json in vault that we want to use--will be concatenated immediately after VAULT_ADDR in a curl)~~ (Removed, now uses a key path directly)
- Adds code to both the release and pr-test actions to run integration tests

### Review Instructions

- **Can't review until the secrets are in GitHub**
- The exact same logic happens on both PR and Release. I think we should be able to make dummy PRs to this branch to trigger the new version of the PR action, though the release action I think we'll need to test in master (maybe? I don't know if GH will let us choose to manually run it from this branch)

### CODE REVIEW

**Comment by jack-r-warren** [comment]
Funny enough, it looks like this branch's actions run on this branch, so the check accordingly fails because there's no secrets yet

File: .github/workflows/pr-test.yml [hunk 1]
```diff
@@ -24,6 +24,44 @@ jobs:
...
-      - name: Unit Test
+      # Get secret from vault
+      # Lifted from https://github.com/broadinstitute/gotc-deploy/'s actions
+      - name: Authenticate to Gcloud
+        id: gcloud
+        env:
+          ROLE_ID: ${{ secrets.ROLE_ID }}
+          SECRET_ID: ${{ secrets.SECRET_ID }}
+          VAULT_ADDR: ${{ secrets.VAULT_ADDR }}
+          SA_KEY_PATH: ${{ secrets.SA_KEY_PATH }}
```
Comment (hjfbynara) [inline .github/workflows/pr-test.yml:35]: generally we don't bother with protecting our paths that we need.  it just makes things difficult to manage and adds a level of indirection that IMO makes things more difficult to read and maintain.  and really the paths themselves are not secret - the contents are.

Comment (jack-r-warren) [reply .github/workflows/pr-test.yml:35]: Fair enough--fixed with [this commit](https://github.com/broadinstitute/push-to-cloud-service/commit/670596a034c9cf75829ef206b1fb3708945b1f6d)

**Comment by jack-r-warren** [comment]
Leaving this as draft until the [gotc-deploy PR](https://github.com/broadinstitute/gotc-deploy/pull/81) goes through to add the SA.

**Pull request merged**